internal/topology/variables/cluster_variable_defaulting.go

```diff
@@ -91,10 +91,9 @@ func defaultClusterVariable(clusterVariable *clusterv1.ClusterVariable, clusterC
 	}
 
 	// Convert schema to Kubernetes APIExtensions schema.
-	apiExtensionsSchema, err := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema)
-	if err != nil {
-		return nil, field.ErrorList{field.Invalid(fldPath, "",
-			fmt.Sprintf("invalid schema in ClusterClass for variable %q: error to convert schema %v", clusterVariable.Name, err))}
+	apiExtensionsSchema, errs := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema, fldPath)
+	if len(errs) > 0 {
+		return nil, errs
 	}
```

Comment:
```suggestion
	apiExtensionsSchema, errs := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema, field.NewPath("schema"))
	if len(errs) > 0 {
		return nil, field.ErrorList{field.Invalid(fldPath, "",
			fmt.Sprintf("invalid schema in ClusterClass for variable %q: error to convert schema %v", clusterClassVariable.Name, errs))}
	}
```

The fldPath here is the path of the variable. I think passing in a new pseudo path is the best we can do. This results in the following error message:
```yaml
[spec.topology.variables[0]: Invalid value: "": invalid schema in ClusterClass for variable "cpu": error to convert schema [schema.default: Invalid value: "1\"": default is not valid JSON: invalid character '"' after top-level value]]
```

Note: we have to use `clusterClassVariable.Name` not `clusterVariable.Name` my previous code would have paniced.

Reply:
It seems strange to be wrapping one error list in another like this. Is the variableName the context you want to preserve here?

Reply:
My main motivation is getting a readable error :)

Without this change it looks like the schema is part of the Cluster. So imho nesting error lists is pretty similar to the usual error nesting and leads to similarly good results.

I would try to get an error which:
* a) express which variable is broken (aka contains the variable path)
* b) tells me which part of the schema is broken (aka tells me something about the location of the error in the schema)

Reply:
The current error looks like this:
```yaml
[spec.topology.variables[0].default: Invalid value: "1\"": default is not valid JSON: invalid character '"' after top-level value]
```
(it points to a non-existing field default on the variable)